### PR TITLE
form-blog-PullRequest

### PR DESCRIPTION
**Pull request by IvonneBenitesRodriguez**
🌷 **About this project**<br/>
This part of the project is about adding forms to our Blog App.

🌷**For it** :<br/>
- I created a featured branch named formBlog
- I created a method called current_user in our ApplicationController to make current user data available to all controllers.
- I created forms to perform the following functions:
   -  Create a Post on behalf of the current_user.
   - Create a comment on behalf of the current_user 
 - Linters were passed.
 - I respected the Gitflow

🌷 **Goal:**<br/>
The goal is to use pre-processed HTML file with embedded Ruby code and use layouts templates for shared content, and in this process feel comfortable about it.

### CODE REVIEW

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      @post.likes_counter + 1)
+
+      redirect_to user_post_path(@user, @post)
+    else
+      redirect_to user_post_path(@user, @post), alert:
+      'Unable to create like.'
+    end
+  end
+
+  private
```
Comment (Munsa1) [inline app/controllers/likes_controller.rb]: - [x] Kindly fix your likes controller,right now when a user likes a comment, the count jumps from 0 to 2 then starts to count normally. Please check the screencast below.

Video
[cinnamon-2023-12-15T234644+0200.webm](https://github.com/IvonneBenitesRodriguez/rails_blog/assets/81932478/0b559acc-482a-4ea2-a934-00cfac52c5f4)

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,12 +1,41 @@
...
+    @post = @user.posts.new(post_params)
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice:
+      'Post created successfully!'
+    else
+      render 'new', alert: 'Unable to create post.'
+    end
+  end
+
```
Comment (Munsa1) [inline app/controllers/posts_controller.rb]: - [x] A post is supposed to be created on behalf of the currently logged in user (current_user) of the app. Kindly do so using the `current_user` method in the application controller

e.g
Sample code
```ruby
def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
```

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    if @comment.save
+      redirect_to user_post_path(@post.author, @post),
+                  notice: 'Comment added!'
+    else
+      render 'posts/show', alert: 'Unable to add
+            comment.'
+    end
+  end
+
```
Comment (Munsa1) [inline app/controllers/comments_controller.rb:21]: - [x] Kindly fix the argument error preventing the user from creating a comment for a post.

image

Video 
[cinnamon-2023-12-16T000124+0200.webm](https://github.com/IvonneBenitesRodriguez/rails_blog/assets/81932478/e267ad00-5370-4ff3-92c1-7811350f81b6)

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @IvonneBenitesRodriguez, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Created Forms :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (IvonneBenitesRodriguez) [reply app/controllers/comments_controller.rb:21]: Dear @Munsa1 
thank you so much for your kind suggestion and explanations,
I appreciate it. I modified the changes you suggested me.
Thanks again, 
Best wishes,

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (IvonneBenitesRodriguez) [reply app/controllers/posts_controller.rb]: Dear @Munsa1 ,
Thank you for your suggestion, 
I modified the changes you suggested,
Best wishes,

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (IvonneBenitesRodriguez) [reply app/controllers/likes_controller.rb]: Dear @Munsa1,
Glad to see you again, 
thank you for your advises, 
I modified as you suggested me, 
Best wishes,

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @IvonneBenitesRodriguez 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on making the requested changes from the previous reviewer
- Excellent communication

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**